### ISSUE TITLE
[a11y] [Text field] Invalid aria-multiline property when using a number

### ISSUE THREAD

**Issue by ddayguerrero** [Shopify/polaris-react#2349]
## Issue summary

While using automated accessibility tools on a `<TextField/>` with a multiline prop (number value), ARIA attributes for multiline do not conform to valid values.

Given the sample [TextField](https://41bbm.csb.app/), `aria-multiline=4` is invalid as per the [wai-aria-1.1](https://www.w3.org/TR/wai-aria-1.1/#aria-multiline) specification

## Expected behavior

ARIA attribute for multiline should be `aria-multiline=true/false` regardless of the `multiline` prop value `boolean | number`.

## Actual behavior

ARIA attribute is currently `aria-multiline={number}` when `multiline` is given an arbitrary `number` value.

## Steps to reproduce the problem

1. Render a Text field with a number value for the multiline prop.
2. Run any A11Y audit tool

## Reduced test case

[Codesandbox](https://codesandbox.io/s/41bbm?module=App.js)

```javascript
function MultilineFieldExample() {
  const [value, setValue] = useState('1776 Barnes Street\nOrlando, FL 32801');

  const handleChange = useCallback((newValue) => setValue(newValue), []);

  return (
    <TextField
      label="Shipping address"
      value={value}
      onChange={handleChange}
      multiline={4}
    />
  );
```

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 4.7.0
- Browser: Chrome
- Device: MBP
- Operating System: MacOS

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```

**Comment by ddayguerrero** [comment]
I'd be more happy to investigate a fix for this!
